File: OOPLearning/app/src/main/java/com/melihakkose/ooplearning/MainActivity.java
package com.melihakkose.ooplearning;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        /*
        User userOne=new User();
        myUser.name="Melih";
        myUser.surname="Akkose";
        myUser.age=22;

        User userTwo= new User();
        myUser.name="Mahmut";
        myUser.surname="Saylan";
        myUser.age=34;
        */

        User userONe=new User("Melih","Akkose",22);

        System.out.println(userONe.name);


        //Encapsulation (Kapsülleme)
        FootballPLayer playerOne=new FootballPLayer("Muhammed Salah","Liverpool FC",32);
        playerOne.setAge(36,"Olymposrec");
        System.out.println(playerOne.getAge());

        playerOne.setAge(50,"wrongpassword");
        System.out.println(playerOne.getAge());

        //Inheritance (Miras Alma)
        LegendsFootballPLayer Ronaldo =new LegendsFootballPLayer("Cristiano Ronaldo","Juventus FC",32);
        System.out.println(Ronaldo.getName());
        System.out.println(Ronaldo.shot());

        //Polymorphism

        //Static Polymorphism
        FootballCoach coach=new FootballCoach();
        System.out.println(coach.name());
        System.out.println(coach.name("Kloop"));
        System.out.println(coach.name("Kloop","Liverpool FC"));

        //Dynamic Polymorphism
        Animal animalOne= new Animal();
        animalOne.sound();

        Dog animalDog=new Dog();
        animalDog.test();
        animalDog.sound();

        //abstract
        System.out.println(userONe.info());

        //Interface
        Guitar myGuiter=new Guitar();
        myGuiter.model="1968 Fender Strat";
        myGuiter.available=true;
        myGuiter.info();





    }
}

File: KotlinOOP/app/src/main/java/com/melihakkose/kotlinoop/MainActivity.kt
package com.melihakkose.kotlinoop

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //var myFootball=Football("Cristiano Ronaldo",50)

        /*
        println(myFootball.name)
        println(myFootball.age.toString())

         */


        /*
        myFootball.age=32
        myFootball.name="Muhammed Salah"
        println(myFootball.name)
        println(myFootball.age.toString())

         */
        /*
        var liverpool=Teams("Liverpool","England",1892,"Premiere League")
        println(liverpool.name)

        //encapsulation
        println(liverpool.returnCategory("Football"))
        println(liverpool.returnCategory("noucamp"))

         */
        /*
        var ronaldo=FootballPlayer("Cristiano Ronaldo","Juventus FC", "Portugal",1897,"Seria A")
        println(ronaldo.returnCategory("Football"))
        ronaldo.cv()
        println(ronaldo.name)
        println("I'm From "+ronaldo.country)
        println("I'm playing at " +ronaldo.team)

         */

        //polymorphism (ayni isimler ile farkli islemler yapmak gibi)

        /*static polymorphism-------------------
        var mathematics=Mathematics()
        println(mathematics.sum())
        println(mathematics.sum(5,5))
        println(mathematics.sum(5,5,5))

        dynamic polymorphism----------------
        var animal=Animal()
        animal.say()
        var tiger=Tiger()
        tiger.test()
        tiger.say()
         */

        //Abstract & Interface
        /*
        var myUser=User()
        println(myUser.info())

         */

        //var myPeople=People() --> Abstract class dan obje olusturulamaz
        /*
        var myShoes=Shoes()
        myShoes.brand="Nike"
        myShoes.stock=true
        println(myShoes.brandName)
        myShoes.info()
        myShoes.info2()

         */

        //Lambda expressions -> fonksiyonu tek satirda yazmayi olanak saglar gibi
        /* BU ORNEGI LAMBDA OLARAK YAZACAGIZ
        fun printString(myString:String){
            println(myString)
        }
        printString("Test String")

         */
        val testString={myString:String->println(myString)}
        testString("Lambda Expressions: Test")

        val multiply={a:Int, b:Int -> a * b}
        println(multiply(5,5))
        val multiply2:(Int,Int)->Int={a,b-> a * b}
        println(multiply2(5,5))

        /*
        Lambda Expressions -> Genelde internetten veri cekerken kullanilir
        asynchrnous -> Senkronize olmayan,
        kullanici ekranini kitlemeden arkaplanda yapilacak islemler icin
        callback function, listener function, completion function
        */
        fun downloadFootballPlayers(url:String, completion:(/*ICINE OBJE DE KOYABILIRDIK*/)-> Unit){
            //url-> Download
            //islemler...
            //OBJEYI OLUSTURUR
            completion(/*FONKSIYONUN ICINE OBJEYI VEREBILIRDIK*/)
        }
        downloadFootballPlayers("transfermarkt.com",{
            println("Completed!")
        })








    }
}